File: frontend/src/types/eventPriceReaction.ts
export interface EventPriceReaction {
  id: string;
  event_id: string;
  asset_id: string;
  return_1d: number | null;
  return_3d: number | null;
  return_5d: number | null;
  return_20d: number | null;
  max_drawdown: number | null;
  volume_change: number | null;
  benchmark_return: number | null;
  excess_return: number | null;
  notes: string | null;
  created_at: string;
}

File: backend/app/schemas/event_price_reaction.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class EventPriceReactionResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    event_id: UUID
    asset_id: UUID
    return_1d: float | None
    return_3d: float | None
    return_5d: float | None
    return_20d: float | None
    max_drawdown: float | None
    volume_change: float | None
    benchmark_return: float | None
    excess_return: float | None
    notes: str | None
    created_at: datetime